src/map/mapgen.rs

```diff
@@ -1,25 +1,129 @@
+
+            let cellularstep = |before: &Vec<u8>, after: &mut Vec<u8>| {
+                for y in 0..(height as isize) {
+                    for x in 0..(width as isize) {
+                        let num_alive = liveneighbor_count(before, x, y);
+                        let index = y as usize * width + x as usize;
+                        if before[index] > liveness_threshold {
+                            after[index] = if num_alive < deathlimit {0u8} else {255u8};
+                        } else {
+                            after[index] = if num_alive < birthlimit {0u8} else {255u8};
+                        }
+                    }
+                }
+            };
+
+            let mut startvec: Vec<u8> = vec![0; height*width];
+            ayn.fill_bytes(startvec.as_mut_slice());
+            let mut scratchvec = vec![0u8; height*width];
+
+            for _ in 1..3 {
+                cellularstep(&startvec, &mut scratchvec);
+                cellularstep(&scratchvec, &mut startvec);
+            }
+
+            for _ in 0..16 {
+                let xd = ayn.next_u64() as usize % width;
+                let yd = ayn.next_u64() as usize % height;
+                let index = yd * width + xd;
+                if startvec[index] < liveness_threshold {
+                    continue;
+                }
+                scratchvec = vec![0u8; height*width];
+
+                let mut filled_count = 0;
+
+                let mut fillstack: Vec<(isize, isize)> = vec!();
+                fillstack.push((xd as isize , yd as isize));
+                let condpush = |x, y, coordstack: &mut Vec<(isize, isize)>| {
+                    if neighborstate(&startvec, x, y) {
+                        coordstack.push((x, y));
+                    }
+                };
+
+                let mut floodfill = |x, y, coordstack: &mut Vec<(isize, isize)>| {
+                    let floodindex = y as usize * width + x as usize;
+                    if scratchvec[floodindex] == 0 {
+                        filled_count += 1;
+                        scratchvec[floodindex] = 255;
+                        condpush(x - 1, y, coordstack);
+                        condpush(x + 1, y, coordstack);
+                        condpush(x, y + 1, coordstack);
+                        condpush(x, y - 1, coordstack);
+                    }
+                };
+
+                while let Some(tuple) = fillstack.pop() {
+                    floodfill(tuple.0, tuple.1, &mut fillstack);
+                }
+                if filled_count > width * height * 2 / 5 {
+                    let gridtiles = scratchvec.drain(..).map(|i| tileset[i as usize].clone()).collect();
+
+                    return Self {
+                        cols: width,
+                        rows: height,
+                        tiles: gridtiles
+                    };
+                }
+            }
+            println!("Grid generation failed! Retrying to generate grid!");
```

Comment:
this should happen silently maybe?

Reply:
Maybe but it's a really exceptional case and it's kind of useful to tell if things might be taking somewhat longer?

Reply:
yes i wasn't sure how often this happened so maybe it's ok but should it at least be printerr or something then?